crates/cheatcodes/src/inspector.rs

```diff
@@ -1974,9 +2009,10 @@ impl Inspector<&mut dyn DatabaseExt> for Cheatcodes {
             ecx.env.tx.gas_price = gas_price;
         }
 
-        if self.startup_zk && !self.use_zk_vm {
-            self.startup_zk = false; // We only do this once.
+        if self.zk_startup_migration.is_allowed() && self.startup_zk && !self.use_zk_vm {
```

Comment:
The `initialize_interp` will never be called more than once, unless the cheatcode inspector is reinitialized...
In this case, having the migration only here will not work if it's initially skipped

Reply:
The `initialize_interp` is called once during a revm execution, but because we keep the inspector ref within the `Executor`, subsequent deploys (i.e. revm executions) are calling it multiple times.

Reply:
If we kept (and used) the ref subsequently, wouldn't state _not_ be commited?

Reply:
clarified offline: state is persisted, but `initialize_interp` is called each executor start, so on external contract deployment (libs, other, test/script) and when invoking test/script functions